src/tribler-common/tribler_common/utilities.py

```diff
@@ -27,6 +29,46 @@ def uri_to_path(uri):
 
 
 fts_query_re = re.compile(r'\w+', re.UNICODE)
+extract_tags_re = re.compile(r'\[\w+\]')
+
+
+@dataclass
+class Query:
+    original_query: Optional[str]
+    tags: Set[str] = field(default_factory=set)
+    fts_text: str = ''
+
+
+def parse_query(query: Optional[str]) -> Query:
+    """
+    The query structure:
+        query = [tag1][tag2] text
+                 ^           ^
+                tags        fts query
+    """
+    if not query:
+        return Query(original_query=query)
+
+    tags = extract_tags(query)
+    fts_text = extract_plain_fts_query_text(query, tags)
+
+    return Query(original_query=query, tags=tags, fts_text=fts_text)
+
+
+def extract_tags(text: Optional[str]) -> Set[str]:
```

Comment:
`text` field could be `None` because it comes from self.top_search_bar.text(): https://github.com/drew2a/tribler/blob/465c7be7b093dd9708d96c3ea1b84198208083a7/src/tribler-gui/tribler_gui/tribler_window.py#L1034

Reply:
Fixed